lib/src/main/java/io/ably/lib/rest/AblyBase.java

```diff
@@ -74,6 +71,11 @@ public AblyBase(ClientOptions options) throws AblyException {
             Log.e(getClass().getName(), msg);
             throw AblyException.fromErrorInfo(new ErrorInfo(msg, 400, 40000));
         }
+        if (options.agents != null && !AblyAgentValidator.areAllValid(options.agents)) {
+            String msg = "invalid agent provided";
+            Log.e(getClass().getName(), msg);
+            throw AblyException.fromErrorInfo(new ErrorInfo(msg, 400, 40000));
```

Comment:
I've based this on the error handling from above's `if(options == null)`, not sure if that's the right way to handle it :thinking:

Reply:
I've asked around and haven't seen any suggestions for alternatives so this generic code suffice.